code/modules/client/preferences2/preferences2.dm

```diff
@@ -0,0 +1,162 @@
+/*
+
+	Preferences 2 - Now with 100% more database
+
+	This system evicts savefiles and uses the database for all playerdata storage.
+
+	All "user" preferences must be stored as tags in the `SS13_preferences` table.
+	Any boolean toggle preference (Show/Hide Deadchat for example) **MUST** be a bitflag toggle stored as a single toggles integer.
+	This is to cut down on the amount of useless columns when you can pack up to 24 binary toggles into a single integer.
+	NOTE: You cant go above 24. BYOND loses precision and you then break everything.
```

Comment:
It's not actually a loss of precision. The shift operator just flat-out won't support shifts of 24 and will return 0, even though floating-point numbers will do that OK. (But it's probably for the best, since to your point if BYOND allowed it, people would try it and then it _would_ lose precision and break everything.)

Reply:
I meant more in the sense of when you manually key in `33554432` and try and shift with it, BYOND cant handle the bitwise checks since it only allocates 24-bits for the pre-decimal place integer area.

*Theres probably more computer science-y terms for this, but this is a DM GitHub project not an assembler.*